docs/tutorials/adding-a-module-to-your-runtime.md

```diff
@@ -0,0 +1,306 @@
+
+> Note: The config attribute only applies to the import directly below it, so in the snippet above, `parity_codec` is always used in both the `std` and `no_std` environments.
+
+### Importing the Contract Module Crate
+
+Okay, now that we have covered the basics of crate features, we can actually import the Contract module. The Contract module is probably the most complicated module in the SRML, so it makes for a good example of some of the trickiness that can be involved when adding additional modules. To give you a hint as to what is to come, you should take a look at the [`Cargo.toml` file for the Contract module](https://github.com/paritytech/substrate/blob/v1.0/srml/contract/Cargo.toml).
+
+First we will add the new dependency by simply copying an existing module, and changing the values. So based on the `balances` import shown above, my `contract import will look like:
+
+```rust
+[dependencies.contract]
+default_features = false
+git = 'https://github.com/paritytech/substrate.git'
+package = 'srml-contract'
+rev = '783ca1892892454e05e234cda5f7a2e42a54461e'
+```
+
+You can see that the Contract module has an `std` feature, thus we need to add the `contract/std` feature to the `std` feature of our runtime:
+
+```rust
+std = [
+    ...
+    'contract/std',
+]
+```
+
+If you forget to set the `contract/std` feature, when building to your native binaries you will get errors like:
+
+```rust
+error: cannot find macro `vec!` in this scope
+   --> /Users/shawntabrizi/.cargo/git/checkouts/substrate-7e08433d4c370a21/783ca18/core/sr-sandbox/src/../without_std.rs:290:24
+    |
+290 |         let mut return_val = vec![0u8; sandbox_primitives::ReturnValue::ENCODED_MAX_SIZE];
+    |                              ^^^
+
+error: aborting due to previous error
+```
+
+If we look closer, it also has another feature: `core`.
+
+**Contract Module `Cargo.toml`**
+```rust
+core = [
+    "wasmi-validation/core",
+]
+```
+
+We will also need to add this feature into our runtime, so we will add the following to our runtime's `Cargo.toml` file:
+
+**contract-chain `Cargo.toml`**
+```rust
+core = [
+	"contract/core",
+]
+```
+
+Thus, when our runtime compiles with the `core` feature, it will use the `contract/core` feature, which will then use the `wasmi-validation/core` feature, and so on.
+
+At this point, you should be able to sanity check that everything compiles correctly with:
+
+```bash
+// Build Wasm binaries with `no_std`
+./scripts/build.sh
+
+// Build native binaries with `std`
+cargo build --release
+```
+
+## Adding the Contract Module
+
+Now that we have successfully imported the Contract module crate, we need to add it to our Runtime.
+
+If you have followed our [other basic tutorials](tutorials/creating-your-first-substrate-chain.md), you would know that we need to implement a `contract::Trait` and also add `Contract: contract::{...}` to our `construct_runtime!` macro.
```

Comment:
```diff
- If you have followed our other basic tutorials, you would know that we need ...
+ If you have followed our other basic tutorials, you may remember that we need ...
```

So it doesn't accidentally sound like, "If you had finished your homework, you would already know this"